File: Darts/src/burlton/dartzee/code/db/VersionEntity.java
package burlton.dartzee.code.db;

import burlton.dartzee.code.utils.DartsDatabaseUtil;

import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.List;

public class VersionEntity extends AbstractEntity<VersionEntity>
{
	private int version = DartsDatabaseUtil.DATABASE_VERSION;
	
	@Override
	public String getTableName() 
	{
		return "Version";
	}

	@Override
	public String getCreateTableSqlSpecific() 
	{
		return "Version int";
	}

	@Override
	public void populateFromResultSet(VersionEntity version, ResultSet rs) throws SQLException 
	{
		version.setVersion(rs.getInt("Version"));
	}

	@Override
	public String writeValuesToStatement(PreparedStatement statement, int i, String statementStr) throws SQLException 
	{
		statementStr = writeInt(statement, i, version, statementStr);
		return statementStr;
	}
	
	public int getVersion()
	{
		return version;
	}
	public void setVersion(int version)
	{
		this.version = version;
	}
	
	public static VersionEntity retrieveCurrentDatabaseVersion()
	{
		List<VersionEntity> entities = new VersionEntity().retrieveEntities("1 = 1");
		if (entities.isEmpty())
		{
			return null;
		}
		
		return entities.get(0);
	}
}

File: Darts/src/burlton/dartzee/code/db/VersionEntity.kt
package burlton.dartzee.code.db

import burlton.desktopcore.code.util.Debug
import burlton.dartzee.code.utils.DartsDatabaseUtil

class VersionEntity : AbstractEntity<VersionEntity>()
{
    var version = DartsDatabaseUtil.DATABASE_VERSION

    override fun getTableName() = "Version"

    override fun getCreateTableSqlSpecific() = "Version INT NOT NULL"

    companion object
    {
        fun retrieveCurrentDatabaseVersion(): VersionEntity?
        {
            val entities = VersionEntity().retrieveEntities("1 = 1")
            return if (entities.isEmpty()) null
            else
            {
                if (entities.size > 1)
                {
                    Debug.stackTrace("Found ${entities.size} rows in Version - should only be 1")
                }

                entities.first()
            }
        }

        fun insertVersion()
        {
            val versionEntity = VersionEntity()
            versionEntity.assignRowId()
            versionEntity.version = DartsDatabaseUtil.DATABASE_VERSION
            versionEntity.saveToDatabase()
        }
    }
}
